web/packages/teleterm/src/ui/ConnectMyComputer/connectMyComputerContext.tsx

```diff
@@ -222,9 +234,42 @@ export const ConnectMyComputerContextProvider: FC<{
     }
   }
 
+  const agentIsNotStarted =
+    currentAction.kind === 'observe-process' &&
+    currentAction.agentProcessState.status === 'not-started';
+
+  useEffect(() => {
```

Comment:
Shouldn't we return early from this `useEffect` if `canUse` is `false`? Otherwise we're going to call `checkIfAgentIsConfigured`, even for clusters where it's irrelevant.

Reply:
It would be great to do it, but it is not that easy.
To get the value of `canUse` we need to fetch data from the auth server, which can take some time.
Until we get that data, `canUse` is  `false`, and it would block calling `checkIfAgentIsConfigured` wich means that we would show to the user the setup page that would switch to status after we unblock the call (in my case it takes 1-2 seconds).

Reply:
Let's document it then that `checkIfAgentIsConfigured` needs to be called irrespective of whether the user actually can use CMC and why.

Matter of fact, I feel like it might be warranted to have two `useEffect`s here, one for `checkIfAgentIsConfigured` and another for autostart. Just so that the next person reading doesn't end up asking the same question as I did. ;)

Reply:
Moved `checkIfAgentIsConfigured` to a separate `useEffect` and added a comment.